libtiff/libtiff_ctypes.py

```diff
@@ -340,17 +345,29 @@ def extender_pyfunc(tiff_struct):
 def add_tags(tag_list):
     tag_list_array = (TIFFFieldInfo * len(tag_list))(*tag_list)
     for field_info in tag_list_array:
-        _name = "TIFFTAG_" + str(field_info.field_name).upper()
-        globals()[_name] = field_info.field_tag
-        if field_info.field_writecount > 1 and field_info.field_type != \
-                TIFFDataType.TIFF_ASCII:
-            tifftags[field_info.field_tag] = (
-                ttype2ctype[
-                    field_info.field_type] * field_info.field_writecount,
-                lambda _d: _d.contents[:])
-        else:
-            tifftags[field_info.field_tag] = (
-                ttype2ctype[field_info.field_type], lambda _d: _d.value)
+        data_t = ttype2ctype[field_info.field_type]
+        convert_c_to_py = lambda d: d.value
+
+        # Handle arrays (except for ASCII arrays aka strings, because they are automatically handled)
+        if (field_info.field_readcount != 1
+            and field_info.field_type != TIFFDataType.TIFF_ASCII
+        ):
+            # Note: typically field_readcount == field_writecount
+            if field_info.field_readcount > 1:
+                data_t = data_t * field_info.field_readcount
+                convert_c_to_py = lambda d: d.contents[:]
+            elif field_info.field_readcount in (TIFF_VARIABLE, TIFF_VARIABLE2):
+                count_t = ctypes.c_uint16 if field_info.field_readcount == TIFF_VARIABLE else ctypes.c_uint32
+                data_t = (count_t, data_t)
+                convert_c_to_py = lambda d: d[1][:d[0]]
+            else:
+                warnings.warn(f"Unsupported readcount {field_info.field_readcount}")
+                # Let's be optimistic and assume the standard behaviour will work
```

Comment:
Thoughts on putting this in a separate function to make the code clearer?

Reply:
Something like `field_info_to_tifftag()`, which converts a single field_info into one tifftag entry? I can try that. It will indeed probably help a bit the readability 🙂